### DISCUSSION TITLE
Are there any plans for a breadcrumb extension with JSON-LD?

### DISCUSSION THREAD

**Discussion by JetGum** [datenstrom/community#494] [Ask a question] [answered]
Hi, 

just thinking that it would be nice if the Breadcrumb extension could automatically add the (Google-preferred) [JSON-LD](https://schema.org/BreadcrumbList) to the breadcrumb output. 

Any thoughts or plans for this?

Cheers,

Carsten

**Comment by GiovanniSalmeri** [comment, marked as answer]
Just out of curiosity (and in order to fall asleep :-) I gave a glance at the specs. Adding the json-ld output seems really easy (also thanks to the magic of the function `json_encode` in PHP). Unless I have made some mistake, it is sufficient to change the function `onParsePageExtra` in `breadcrumb.php` like this:

```
    // Handle page extra data
    public function onParsePageExtra($page, $name) {
        if ($name=="header") {
            $bc = [
                "@context" => "http://schema.org",
                "@type" => "BreadcrumbList",
                "itemListElement" => [],
            ];
            $pages = $this->yellow->content->path($page->getLocation(true), true);
            foreach ($pages as $n => $pageBreadcrumb) {
                $bc["itemListElement"][] = [
                    "@type" => "ListItem",
                    "position" => $n + 1,
                    "item" => [
                        "@id" => $pageBreadcrumb->getLocation(true),
                        "name" => $pageBreadcrumb->get("titleNavigation"),
                    ],
                ];
            }
            return "<script type=\"application/ld+json\">".json_encode($bc)."</script>\n";
        } else {
            return $this->onParseContentShortcut($page, $name, "", "block");
        }
    }
```

**Comment by JetGum** [comment]
@GiovanniSalmeri 

Thx a million, this works like a charm. Implemented on [https://www.aussenpool.de/](url) and teste with [Google](https://search.google.com/structured-data/testing-tool?hl=de#url=https%3A%2F%2Fwww.aussenpool.de%2Fpool-wasserpflege%2Fpoolwasser-gruen-was-tun%2F).

Cheers!

Carsten

PS: The changes will be overwritten when breadcrumbs.php is updated, right?

**Comment by GiovanniSalmeri** [comment]
> PS: The changes will be overwritten when breadcrumbs.php is updated, right?

Yes, right. But I could wrap this addition in a separate tiny extension (let's say `breadcrumb-schema`, for example), so it wouldn't be overwritten. Unless it could be added directly in the official `breadcrumb` extension, in a manner similar to `sitemap`, which creates both a sitemap page and the XML which Google likes. But honestly I have no idea whether this would be really useful for most users, I like very much Yellow also because it is extremely light :-) @markseu Your opinion?

**Comment by markseuffert** [comment]
I agree with Giovanni. I would put it into a separate extension. It doesn't take much time to make an extension. Developers are always welcome, Carsten.

**Comment by JetGum** [comment]
Hey all, I wouldn't dare to call myself a developer :-) still, I'll check the docs to see if I can get my head around this. I would use Giovanni's code anyway. Cheers,

Carsten

PS: @GiovanniSalmeri I agree with you completely. Yellow is extremely light, and I have a pagespeed of 96-99 on Google Pagespeed Insights. That is mainly due to the code not being bloated and using only the plugins / JS/ CSS you need. Still, I would love to see some more SEO or Affiliate plugins come to light... :-)

**Comment by annaesvensson** [comment]
Thanks for your feedback. I am afraid to say that we don't support JSON-LD, not unless there is more demand. We can't promise if we may add something like this later. However we definitely appreciate the feedback. If you want to make a new extension, [go ahead and let us know if you need help](https://datenstrom.se/yellow/help/contributing-guidelines).
